backend/coins/eth/account.go

```diff
@@ -320,43 +320,42 @@ func (account *Account) newTx(
 	}
 
 	var value *big.Int
-	var gasLimit uint64
 	if amount.SendAll() {
-		// The value is set below after determining the fee.
-		gasLimit = 21000 // Hardcoded to prevent cyclic gas estimation.
+		value = account.balance.BigInt() // set here only temporarily to estimate the gas
 	} else {
 		allowZero := true
 		parsedAmount, err := amount.Amount(big.NewInt(params.Ether), allowZero)
 		if err != nil {
 			return nil, err
 		}
 		value = parsedAmount.BigInt()
+	}
 
-		address := common.HexToAddress(recipientAddress)
-		message := ethereum.CallMsg{
-			From:     account.address.Address,
-			To:       &address,
-			Gas:      0,
-			GasPrice: suggestedGasPrice,
-			Value:    value,
-			Data:     data,
-		}
-
-		gasLimit, err = account.coin.client.EstimateGas(context.TODO(), message)
-		if err != nil {
-			account.log.WithError(err).Error("Could not estimate the gas limit.")
-			return nil, errp.WithStack(coin.ErrInvalidData)
-		}
+	address := common.HexToAddress(recipientAddress)
+	message := ethereum.CallMsg{
+		From:     account.address.Address,
+		To:       &address,
+		Gas:      0,
+		GasPrice: suggestedGasPrice,
+		Value:    value,
+		Data:     data,
+	}
+	gasLimit, err := account.coin.client.EstimateGas(context.TODO(), message)
+	if err != nil {
+		account.log.WithError(err).Error("Could not estimate the gas limit.")
+		return nil, errp.WithStack(coin.ErrInvalidData)
 	}
 
 	fee := new(big.Int).Mul(new(big.Int).SetUint64(gasLimit), suggestedGasPrice)
 
 	if amount.SendAll() {
+		// Set the value correctly and check that the fee is smaller than or equal to the balance.
 		value = new(big.Int).Sub(account.balance.BigInt(), fee)
-		if value.Sign() <= 0 {
+		if value.Sign() < 0 {
```

Comment:
`<=` was with the assumption of not having zero value transfers. Since the `allowZero` PR, this fix makes sense :+1: